src/Raven.Server/Documents/Handlers/HugeDocumentsHandler.cs

```diff
@@ -0,0 +1,55 @@
+using System.Globalization;
+using System.Threading.Tasks;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class HugeDocumentsHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/huge-documents", "GET")]
+        public Task HugeDocuments()
+        {
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            using (context.OpenReadTransaction())
+            {
+                writer.WriteStartArray();
+
+                var isFirst = true;
+
+                foreach (var key in context.DocumentDatabase().HugeDocuments.GetHugeDocuments())
```

Comment:
Why do it like this? You are forcing us to do a LOT of lookups into this.
You can return both key & size at the same time